composer/core/time.py

```diff
@@ -212,6 +213,30 @@ def from_duration(cls, duration: float) -> Time:
         """
         return cls(duration, TimeUnit.DURATION)
 
+    @classmethod
+    def from_timedelta(cls, timestring: str) -> Time:
+        """Create a :class:`Time` with units of :attr:`TimeUnit.SECOND`.
+
+        Equivalent to ``Time(batch, TimeUnit.SECOND)``.
+
+        Args:
+            timestring (int): timedelta string in _h_m_s.
+
+        Returns:
+            Time: :class:`Time` instance, in seconds.
+        """
+        # Convert timestring to be strptime parsable
+        if 'h' not in timestring:
```

Comment:
I would prefer explicit validation first:
```
def verify_timestamp(timestamp):
    # Define the regex pattern
    pattern = r'^(\d+h)?(\d+m)?(\d+s)?$'
    
    # Match the pattern against the timestamp
    match = re.match(pattern, timestamp)
    
    # Return True if it matches, otherwise False
    return bool(match)
```